test/unit-test/ice_api/ice_api_utest.c

```diff
@@ -19,13 +87,273 @@
+                       result );
+
+    initInfo.cryptoFunctions.hmacFxn = testHmacFxn;
+    initInfo.creds.pLocalUsername = NULL;
+
+    result = Ice_Init( &( context ),
+                       &( initInfo ));
+
+    TEST_ASSERT_EQUAL( ICE_RESULT_BAD_PARAM,
+                       result );
+
+    initInfo.creds.pLocalUsername = &( buffer[ 0 ] );
+    initInfo.creds.pLocalPassword = NULL;
+
+    result = Ice_Init( &( context ),
+                       &( initInfo ));
+
+    TEST_ASSERT_EQUAL( ICE_RESULT_BAD_PARAM,
+                       result );
+
+    initInfo.creds.pLocalPassword = &( buffer[ 0 ] );
+    initInfo.creds.pRemoteUsername = NULL;
+
+    result = Ice_Init( &( context ),
+                       &( initInfo ));
+
+    TEST_ASSERT_EQUAL( ICE_RESULT_BAD_PARAM,
+                       result );
+
+    initInfo.creds.pRemoteUsername = &( buffer[ 0 ] );
+    initInfo.creds.pRemotePassword = NULL;
+
+    result = Ice_Init( &( context ),
+                       &( initInfo ));
+
+    TEST_ASSERT_EQUAL( ICE_RESULT_BAD_PARAM,
+                       result );
+
+    initInfo.creds.pRemotePassword = &( buffer[ 0 ] );
+    initInfo.creds.pCombinedUsername = NULL;
+
+    result = Ice_Init( &( context ),
+                       &( initInfo ));
+
+    TEST_ASSERT_EQUAL( ICE_RESULT_BAD_PARAM,
+                       result );
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Validate ICE Init functionality.
+ */
+void test_iceInit( void )
+{
+    IceContext_t context;
+    IceInitInfo_t initInfo = { 0 };
+    IceResult_t result;
+    IceCandidate_t localCandidateArray[ 10 ], remoteCandidateArray[ 10 ];
+    IceCandidatePair_t candidatePairArray[ 100 ];
+    TransactionIdStore_t transactionIdStore[ 16 ];
+
+    initInfo.pLocalCandidatesArray = &( localCandidateArray[ 0 ] );
+    initInfo.pRemoteCandidatesArray = &( remoteCandidateArray[ 0 ] );
+    initInfo.pCandidatePairsArray = &( candidatePairArray[ 0 ] );
+    initInfo.pStunBindingRequestTransactionIdStore = &( transactionIdStore[ 0 ] );
+    initInfo.cryptoFunctions.randomFxn = testRandomFxn;
+    initInfo.cryptoFunctions.crc32Fxn = testCrc32Fxn;
+    initInfo.cryptoFunctions.hmacFxn = testHmacFxn;
+    initInfo.creds.pLocalUsername = ( uint8_t * ) "localUsername";
+    initInfo.creds.pLocalPassword = ( uint8_t * ) "localPassword";
+    initInfo.creds.pRemoteUsername = ( uint8_t * )  "remoteUsername";
+    initInfo.creds.pRemotePassword = ( uint8_t * ) "remotePassword";
+    initInfo.creds.pCombinedUsername = ( uint8_t * ) "combinedUsername";
+    initInfo.isControlling = 1;
+
+    result = Ice_Init( &( context ),
+                       &( initInfo ));
+
+    TEST_ASSERT_EQUAL( ICE_RESULT_OK,
+                       result );
+    TEST_ASSERT_EQUAL( localCandidateArray,
+                       context.pLocalCandidates );
+    TEST_ASSERT_EQUAL( remoteCandidateArray,
+                       context.pRemoteCandidates );
```

Comment:
`maxRemoteCandidates` and `numRemoteCandidates` should also be checked.

Reply:
Will add, by default both will be 0.